Index: setup.py
===================================================================
--- setup.py	(revision 620)
+++ setup.py	(working copy)
@@ -3,6 +3,7 @@
 import os
 import sys
 from setuptools import setup, Extension
+from setup_jaraco_mysql import setup
 
 if sys.version_info < (2, 3):
     raise Error("Python-2.3 or newer is required")
Index: setup_jaraco_mysql.py
===================================================================
--- setup_jaraco_mysql.py	(revision 0)
+++ setup_jaraco_mysql.py	(revision 0)
@@ -0,0 +1,24 @@
+#!/usr/bin/env python
+
+"""
+Wrap setuptools.setup to remove the _mysql and _mysql_exceptions
+modules from the build and add a dependency on jaraco.mysql.
+"""
+
+import sys
+from setuptools import setup as setuptools_setup
+
+def jaraco_mysql_setup(**metadata):
+	# remove extension modules
+	del metadata['ext_modules']
+	metadata['name'] = 'mysqldb-jaraco'
+	metadata['py_modules'].remove('_mysql_exceptions')
+	# add jaraco.mysql to the required modules
+	metadata.setdefault('install_requires', []).append('jaraco.mysql')
+	setuptools_setup(**metadata)
+
+try:
+	sys.argv.remove('--use-jaraco-mysql')
+	setup = jaraco_mysql_setup
+except ValueError:
+	setup = setuptools_setup

Property changes on: setup_jaraco_mysql.py
___________________________________________________________________
Added: svn:keywords
   + Id Rev Author Date

Index: setup_windows.py
===================================================================
--- setup_windows.py	(revision 620)
+++ setup_windows.py	(working copy)
@@ -1,12 +1,31 @@
+
+import _winreg as winreg
+from itertools import count
+
+def registry_key_subkeys(key):
+    for index in count():
+        try:
+            yield winreg.EnumKey(key, index)
+        except WindowsError:
+            break
+
+def get_mysql_root(options):
+    mySQLKey = winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE, r"SOFTWARE\MySQL AB")
+    installedVersions = registry_key_subkeys(mySQLKey)
+    latestServerKeyName = sorted(installedVersions)[-1]
+    serverKeyName = options.get('registry_key', latestServerKeyName)
+    serverKey = winreg.OpenKey(mySQLKey, serverKeyName)
+    mysql_root, dummy = winreg.QueryValueEx(serverKey,'Location')
+    return mysql_root
+
 def get_config():
-    import os, sys, _winreg
+    import os, sys
     from setup_common import get_metadata_and_options, enabled, create_release_file
 
     metadata, options = get_metadata_and_options()
 
-    serverKey = _winreg.OpenKey(_winreg.HKEY_LOCAL_MACHINE, options['registry_key'])
-    mysql_root, dummy = _winreg.QueryValueEx(serverKey,'Location')
-
+    mysql_root = get_mysql_root(options)
+    
     extra_objects = []
     static = enabled(options, 'static')
     # XXX static doesn't actually do anything on Windows
Index: tests/test_MySQLdb_capabilities.py
===================================================================
--- tests/test_MySQLdb_capabilities.py	(revision 620)
+++ tests/test_MySQLdb_capabilities.py	(working copy)
@@ -12,6 +12,15 @@
     connect_args = ()
     connect_kwargs = dict(db='test', read_default_file='~/.my.cnf',
                           charset='utf8', sql_mode="ANSI,STRICT_TRANS_TABLES,TRADITIONAL")
+
+    import re, os
+    # grab parameters from a sqlalchemy-style URI out of the
+    #  environment.
+    uri_pattern = re.compile('mysql://(?P<user>.*):(?P<passwd>.*)@(?P<host>.*)/(?P<db>.*)')
+    uri_match = uri_pattern.match(os.environ.get('MYSQL_TEST_DB'))
+    if uri_match:
+		connect_kwargs.update(uri_match.groupdict())
+
     create_table_extra = "ENGINE=INNODB CHARACTER SET UTF8"
     leak_test = False
     
Index: site.cfg
===================================================================
--- site.cfg	(revision 620)
+++ site.cfg	(working copy)
@@ -13,6 +13,5 @@
 #mysql_config = /usr/local/bin/mysql_config
 
 # The Windows registry key for MySQL.
-# This has to be set for Windows builds to work.
-# Only change this if you have a different version.
-registry_key = SOFTWARE\MySQL AB\MySQL Server 5.0
+# Override this if the latest version installed is not adequate
+#registry_key = SOFTWARE\MySQL AB\MySQL Server 5.0
Index: MySQLdb/cursors.py
===================================================================
--- MySQLdb/cursors.py	(revision 620)
+++ MySQLdb/cursors.py	(working copy)
@@ -168,6 +168,8 @@
                 self.errorhandler(self, TypeError, m)
         except:
             exc, value, tb = exc_info()
+            #import traceback
+            #traceback.print_exc()
             del tb
             self.messages.append((exc, value))
             self.errorhandler(self, exc, value)
